### DISCUSSION TITLE
Osclass 5.0.0

### DISCUSSION THREAD

**Discussion by navjottomer** [mindstellar/shopclass#335] [Q&A]
# Osclass 5.0.0 Changelog:

* New geodata support from https:/github.com/mindstellar/geodata 100+ new countries with updated cities and regions
* New Osclass installer boostrap5 based UI.
* New order by relevance option in search.
* New templates for Admin-dashboard login templates.
* New Osclass auto-updater, you can upgrade to prerelease version if you add this in config.php define('
  ENABLE_PRERELEASE', true);
* New You can define a new maintenance template by just placing the maintenance.php file in your theme, see
  mindstellar/theme-bender@8faac8d
* New PHP classes for osclass core functionality for better performance, security and compatibility.
* New JS enqueue methods now load your script in the footer if enqueued after header hook.
* New compatibility improvement to MySQL-8, PHP8.0, PHP7.4 test are passing for PHP nightly too.
* Significant MySql queries reduction in Search Modal.
* Removed osclass Multisite a better alternative will be provided in future.
* Fix bug of listing invalid themes in Admin Appearance
* Fix PHP notices while saving Admin settings.
* Restructured whole osclass core.
* Core osclass now using autoloader for classes and external libraries.
* More option in image text watermark in media settings.
* Multiple security vulnerabilities are fixed.
* Updated breadcrumb Schema
* Now languages can be imported in via our repositories.
* jQuery updated to latest 3.5.x branch with other JS libraries, TinyMCE is updated to 5.x branch.

See commit history for changelog.

Source: https://github.com/mindstellar/Osclass

<hr /><em>This discussion was created from the release <a href='https://github.com/mindstellar/Osclass/releases/tag/5.0.0'>Osclass 5.0.0</a>.</em>

**Comment by nootkan** [comment]
So I am confused.  OsclassPoint claims they are in charge of osclass upgrades so I downloaded 4.4 and now I see you are back with an upgrade to 5.0.  Can I still upgrade to your version of osclass or will it conflict with OsclassPoint's version?

**Reply by navjottomer** [reply]
I can't say much about them, they can claim anything. It's just bad to get a open source project and develop it behind the door. They could have helped here on this project but they didn't due to some sloppy reasons.

BTW Upgrading from osclasspoint release may still be possible, except few issues.
Just make a backup before upgrading. Let me know about it.

**Reply by buninsan** [reply]
Hello
unfortunately I get errors in the new version of the script:

When adding an ad
item-post

Notice: Undefined index: fk_i_category_id in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 590

Notice: Undefined index: pk_i_id in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 600

Notice: Undefined index: s_city in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 601

Notice: Undefined index: s_title in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 607

Notice: Undefined index: s_secret in /sites/site.com/oc-includes/osclass/emails.php on line 599

Notice: Undefined index: pk_i_id in /sites/site.com/oc-includes/osclass/emails.php on line 599

Notice: Undefined index: s_secret in /sites/site.com/oc-includes/osclass/emails.php on line 600

Notice: Undefined index: pk_i_id in /sites/site.com/oc-includes/osclass/emails.php on line 600

Notice: Undefined index: pk_i_id in /sites/site.com/oc-includes/osclass/emails.php on line 616

Notice: Undefined index: s_contact_name in /sites/site.com/oc-includes/osclass/emails.php on line 617

Notice: Undefined index: s_contact_email in /sites/site.com/oc-includes/osclass/emails.php on line 618

Notice: Undefined index: s_title in /sites/site.com/oc-includes/osclass/emails.php on line 619

Notice: Undefined index: fk_i_category_id in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 590

Notice: Undefined index: pk_i_id in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 600

Notice: Undefined index: s_city in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 601

Notice: Undefined index: s_title in /sites/site.com/oc-includes/osclass/helpers/hDefines.php on line 607

Notice: Undefined index: s_contact_email in /sites/site.com/oc-includes/osclass/emails.php on line 647

Notice: Undefined index: s_contact_name in /sites/site.com/oc-includes/osclass/emails.php on line 648

Fatal error: Uncaught PHPMailer\PHPMailer\Exception: Invalid address: (to): in /sites/site.com/oc-includes/vendor/phpmailer/phpmailer/src/PHPMailer.php:1092 Stack trace: #0 /sites/site.com/oc-includes/vendor/phpmailer/phpmailer/src/PHPMailer.php(1014): PHPMailer\PHPMailer\PHPMailer->addOrEnqueueAnAddress('to', '', '') #1 /sites/site.com/oc-includes/osclass/utils.php(347): PHPMailer\PHPMailer\PHPMailer->addAddress('', NULL) #2 /sites/site.com/oc-includes/osclass/emails.php(654): osc_sendMail(Array) #3 /sites/site.com/oc-includes/osclass/classes/Plugins.php(412): fn_email_new_item_non_register_user(Array) #4 /sites/site.com/oc-includes/osclass/helpers/hPlugins.php(46): Plugins::runHook('hook_email_new_...', Array) #5 /sites/site.com/oc-includes/osclass/classes/actions/ItemActions.php(613): osc_run_hook('hook_email_new_...', Array) #6 /sites/site.com/oc-includes/osclass/classes/actions/ItemActions.php(318): ItemA in /sites/site.com/oc-includes/vendor/phpmailer/phpmailer/src/PHPMailer.php on line 1092

**Reply by navjottomer** [reply]
What plugin? theme ?

**Reply by buninsan** [reply]
New theme by defected.dev

**Reply by navjottomer** [reply]
Although I didn't encounter any of these issues, I think there are places where we can harden our empty array cases.

**Reply by buninsan** [reply]
I understood why I got errors when updating the script.
There is no button in the admin panel to update the database, I had to follow the link and the errors disappeared.
http://www.example.com/oc-admin/index.php?page=upgrade&action=upgrade-funcs
